plugins/common/proxy/proxy.go

```diff
@@ -4,21 +4,56 @@ import (
 	"fmt"
 	"net/http"
 	"net/url"
+
+	"golang.org/x/net/proxy"
 )
 
 type HTTPProxy struct {
+	UseProxy     bool   `toml:"use_proxy"`
```

Comment:
Can we rename `UseProxy` to `UseSystemProxy`? and `use_proxy` to `use_system_proxy`